### PR TITLE
DeadBranchElim: Slightly more defensive coding

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

**Comment by dneto0** [comment]
This addresses my minor concerns from #863 

Please take a look @greg-lunarg

**Comment by greg-lunarg** [comment]
Looks good.

My take away is that we need to document assumptions based on the spec using asserts.

Among other benefits, it future proofs the code from changes to the spec.

**Comment by dneto0** [comment]
> My take away is that we need to document assumptions based on the spec using asserts.

I don't intend that to be a hard-and-fast rule.  It's a judgement call, based on readability concerns.  Some things are easier to remember than others.  The asserts or comments are an assist to the reader, and carry some redundancy. 

Code review is for correctness *and* readability.  Unfortunately readability is fuzzy and subjective.   So let's keep talking.  :-)

File: source/opt/dead_branch_elim_pass.cpp [hunk 1]
```diff
@@ -118,6 +118,7 @@ bool DeadBranchElimPass::GetConstInteger(uint32_t selId, uint32_t* selVal) {
   ir::Instruction* sInst = def_use_mgr_->GetDef(selId);
   uint32_t typeId = sInst->type_id();
   ir::Instruction* typeInst = def_use_mgr_->GetDef(typeId);
```
Comment (atgoo) [inline source/opt/dead_branch_elim_pass.cpp]: assert(typeInst);

Comment (dneto0) [reply source/opt/dead_branch_elim_pass.cpp]: Point taken.  I changed the following test to be

if (!typeInst || (typeInst->opcode() != SpvOpTypeInt)) return false;

**Comment by dneto0** [comment]
Rebased. @atgoo please take another look

**Comment by dneto0** [comment]
Hi @atgoo can you take another look?

**Comment by dneto0** [comment]
Rebased

**Pull request merged**